### ISSUE TITLE
docker-compose build leads to error 

### ISSUE THREAD

**Issue by rrbn** [lieblb/testilias#1]
first build leads to the following error: 
 
WARNING: The EMBEDDED_ILIAS_PORT variable is not set. Defaulting to a blank string.
WARNING: The ILIASTEST_ARGUMENTS variable is not set. Defaulting to a blank string.
WARNING: The TILTR_PORT variable is not set. Defaulting to a blank string.
ERROR: The Compose file './docker-compose.yml' is invalid because:
services.web.ports is invalid: Port ranges don't match in length
services.master.ports is invalid: Port ranges don't match in length

**Comment by lieblb** [comment]
Did you use `docker-compose build` directly  (won't work) or the packaged `./compose.py up` (should work)?

`docker-compose build` alone does not work, since the docker file expects the listed environment variables to be set, which is done in `compose.py` (the latter also does some additional magic to make containers known to each other).

In order to use `docker compose` directly, you'd currently need to do

```
TILTR_PORT=11150 EMBEDDED_ILIAS_PORT=11145 ILIASTEST_ARGUMENTS="--ilias-url http://web:80/ILIAS?client_id=ilias --ilias-admin-user root --ilias-admin-password odysseus --ext-ilias-port 11145 --tiltr-port 11150" docker-compose build
````

Still, the latter won't work for running for running TiltR at the moment (only for starting up the interface and running the embedded ILIAS).

**Comment by lieblb** [comment]
I've addressed this with 49714e3f4d2ca853ffd09c75922802c66ddcb291. You can now use `docker-compose build` and `docker-compose up` and TiltR will work.

**Comment by rrbn** [comment]
Thanks alot !
